pkg/spiffe/spiffe.go

```diff
@@ -36,23 +34,8 @@ func DetermineTrustDomain(commandLineTrustDomain string, isKubernetes bool) stri
 }
 
 // GenSpiffeURI returns the formatted uri(SPIFFEE format for now) for the certificate.
-func GenSpiffeURI(ns, serviceAccount string) (string, error) {
-	var err error
-	if ns == "" || serviceAccount == "" {
```

Comment:
Why remove this check?

Reply:
AFAIK this is unneeded. We can still generate a Spiffe URI without them, and everything works fine. We actually do generate this without serviceAccount often, as some kube-system pods don't have a service account, and it doesn't seem to cause any issues other than spamming the logs. We could just not log it, but if we are not logging it and there aren't any issues, then why not just remove the check? See #12108 for more details

Reply:
No service account definitely means something wrong, and we need to log it, to error level if there is too much spamming. But at the end of they day, the right fix is to figure out why pilot does not have service account. 

As Oliver mentioned, spiffe://cluster.local/ns/kube-system/sa/ is the wrong format, we should detect it.

Reply:
We should not generate a certificate with empty service account. Please directly return an error when the service account is empty. We may make the log in debug level if there are always pods that don't have service accounts.